core/src/test/java/com/adobe/aem/commons/assetshare/README.md

```diff
@@ -0,0 +1,156 @@
+
+Test method naming:
+* Shares the name of method being tested, ex: `getTitle()`
+* Prefixed with `test`, ex: the test method is `testGetTitle()`
+* Optionally post-fixed with an underscore-prefixed, capitalized condition name, ex: `testGetTitle__EmptyTitle()`
+
+For example, to test the method `getTitle()` in `HelloWorld.java`:
+
+   `String getTitle()`
+
+The following test methods are created in `HelloWorldImplTest.java`:
+
+    `@Test void testGetTitle()` which tests for the normal, populated title condition
+    `@Test void testGetTitle_EmptyTitle()`, which tests for the condition where the title is missing
+
+### Resource files (JSON)
+
+AEM Mocks provide support to load JSON into mock AEM context and write tests against that, avoiding manual mocking of resources, OSGi services, etc. using AEM-agnostic tools like Mockito.
+
+The following convention for defining JSON resource files should be followed:
+
+* JSON files are created under test/resources under a folder structure that follows the test case class's package and file name, for example:
+
+    `core/src/test/resources/com/adobe/aem/commons/assetshare/components/impl/HelloWorldImplTest.json`
+
+provides the JSON required by the test case `core/src/test/java/com/adobe/aem.commons/assetshare/components/impl/HelloWorldImplTest.java`.
+
+Each test case's JSON file can define multiple, well-named JSON definition roots that represent different states and can, in turn, be used across one or many test methods in the corresponding test case.
+
+JSON definitions should contain the MINIMUM definition to satisfy the test-cases. This means NO unnecessary junk data from JCR exports, for example, jcr:lastModifiedBy, cq:lastReplicated, cq:responsive, etc. nodes/properties unless they are required by a test.
+
+For example, the HelloWorldImplTest.java may need to test for: a fully configured ("default")  component, a wholly un-configured component ("empty") and a component with ONLY the title populated ("title") using the JSON file defined below:
+
+```
+{
+  "default": {
+      jcr:primaryType: "nt:unstructured",
+      sling:resourceType: "asset-share-commons/components/search/hello-world"
+      title: "hello world!",
+      propX: "foo"
+  },
+  "empty": {
+      jcr:primaryType: "nt:unstructured",
+      sling:resourceType: "asset-share-commons/components/search/hello-world"
+  },
+  "title": {
+      jcr:primaryType: "nt:unstructured",
+      sling:resourceType: "asset-share-commons/components/search/hello-world"
+      title: "hello world!",
+  }
+```
+
+#### Example consumption of JSON
+
+```
+@Before
+public void setup() {
+   ctx.load().json("HelloWorldImplTest.json");
+}
+
+...
+
+@Test
+public void testGetTitle__EmptyTitle()  {
+    context.currentResource("/empty");
+    ...
+}
+```
+
+## A sample is worth 1000 words
+
+
+### Sample test case
+
+`core/src/test/java/com/adobe/aem/commons/assetshare/components/impl/HelloWorldImplTest.java`
+
+```
```

Comment:
Do we want to add the package imports to the sample? Might help with potential same-name collisions...?